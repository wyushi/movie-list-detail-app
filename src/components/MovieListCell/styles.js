import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  touchableWrapper: {

  },
  container: {
    flexDirection: 'row',
    backgroundColor: '#fff',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  thumbnail: {
    height: 138,
    width: 92,
    backgroundColor: '#888'
  },
  info: {
    flex: 1,
    marginLeft: 8,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold'
  },
  releaseDate: {
    color: 'grey'
  }
})
