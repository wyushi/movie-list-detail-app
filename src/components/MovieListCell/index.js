import React from 'react'
import { View, Text, Image } from 'react-native'

import styles from './styles'

const MovieListCell = (props) => {
  
  const { movie={} } = props
  return (
    <View style={styles.container}>
      <Image 
        style={styles.thumbnail}
        source={{uri: 'https://image.tmdb.org/t/p/w92/' + movie['poster_path']}}
      />
      <View style={styles.info}>
        <Text style={styles.title}>{movie['title']}</Text>
        <Text style={styles.releaseDate}>{movie['release_date']}</Text>
      </View>
    </View>
  )
}

export default MovieListCell