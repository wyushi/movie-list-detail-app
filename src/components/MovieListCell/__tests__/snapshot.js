import React from 'react'
import renderer from 'react-test-renderer'

import MovieListCell from '../index'
import data from './data.json'

it('renders a MovieListCell using Snapshots', () => {
  expect(renderer.create(
    <MovieListCell
      movie={data.results[0]}
      onCellPress={_ => {}}
    />
  )).toMatchSnapshot();
})