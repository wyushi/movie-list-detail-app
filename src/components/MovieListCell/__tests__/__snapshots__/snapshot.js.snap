// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a RepoItem using Snapshots 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w342//5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg",
        }
      }
      style={
        Object {
          "backgroundColor": "#888",
          "height": 138,
          "width": 92,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Aquaman
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
          }
        }
      >
        2018-12-07
      </Text>
    </View>
  </View>
</View>
`;
